src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -0,0 +1,170 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+syntax = "proto3";
+
+package aspire.v1;
+
+import "google/protobuf/timestamp.proto";
+
+////////////////////////////////////////////
+
+message ApplicationInformationRequest {
+}
+
+message ApplicationInformationResponse {
+    string application_name = 1;
+    string application_version = 2;
+}
+
+////////////////////////////////////////////
+
+message ResourceCommandRequest {
+    string command_type = 1;
+    ResourceId resource_id = 2;
+    optional string confirmation_message = 3;
```

Comment:
Interesting idea. I think it could make sense. Maybe using the [`google.protobuf.Value`](https://learn.microsoft.com/en-us/aspnet/core/grpc/protobuf?view=aspnetcore-8.0#value) James pointed to earlier.

It'd be possible to add this later safely too. Do you have a specific use case in mind? That might help prioritize.

Reply:
I added a parameter to the commands.

Reply:
E.g. if we want to allow dashboard users to scale resources up/down, the parameter might be desired number of replicas.

Reply:
Having user-editable parameters would be more complex, as we then need to add metadata for the UI to know how to display/constrain/validate the parameter's value. Before we ship V1 I think we'll want to revisit commands and see whether to keep parameters (and whether to keep the command support at all). It'll depend how much utility we can get out of them between now and then.